gen/goav2gen/helpers.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"mime"
+	"regexp"
+	"strings"
+
+	"bitbucket.org/pkg/inflect"
+	"github.com/rightscale/rsc/gen"
+)
+
+var (
+	// Capture all alphanumerical parts to build go identifier from raw param name
+	partsRegexp = regexp.MustCompile("[^[:alnum:]]+")
+
+	// Check whether string only contains blank characters
+	blankRegexp = regexp.MustCompile(`^\s*$`)
+)
+
+// Produce action return type name
+func toGoReturnTypeName(name string, slice bool) string {
+	slicePrefix := ""
+	if slice {
+		slicePrefix = "[]"
+	}
+	return fmt.Sprintf("%s*%s", slicePrefix, toGoTypeName(name))
+}
+
+// toGoTypeName converts from swagger/goa type names to go type names
+func toGoTypeName(name string) string {
+	switch name {
+	case "string":
+		return "string"
+	case "integer":
+		return "int"
+	case "boolean":
+		return "bool"
+	}
+	return name
+
+}
+
+// Parse native names into go parameter names
+func toVarName(name string) string {
+	p := partsRegexp.ReplaceAllString(name, "_")
+	return inflect.CamelizeDownFirst(p)
+}
+
+var toMethodName = inflect.Camelize
+var toTypeName = inflect.Camelize
+
+// Return dumpable representation of given object
+func prettify(o interface{}) string {
+	s, err := json.MarshalIndent(o, "", "    ")
+	if err != nil {
+		return fmt.Sprintf("%+v", o)
+	}
+	return string(s)
+}
+
+// cleans up description, removing blank lines and extraneous info
+func cleanDescription(doc string) string {
```

Comment:
It might be useful to have the required security scopes in showing help.

Reply:
Yeah could go either way -- I think I'd rather link to a page describing the roles/scopes as a whole maybe as a blurb at the top, the rsc --help command is a bit too cluttered right now with every product thrown in at the same time.